src/test/java/org/gridsuite/shortcircuit/server/repositories/FaultResultRepositoryTest.java

```diff
@@ -0,0 +1,208 @@
+                List.of(faultResultEntity1, faultResultEntity2, faultResultEntity3)),
+            Arguments.of(
+                resultMagnitudeEntity,
+                List.of(
+                    new ResourceFilter(ResourceFilter.DataType.TEXT, ResourceFilter.Type.EQUALS, null, "limitViolations.limitType")),
+                List.of())
+        );
+    }
+
+    private Stream<Arguments> provideContainsNestedFieldsFilters() {
+        return Stream.of(
+            Arguments.of(
+                resultMagnitudeEntity,
+                List.of(
+                    new ResourceFilter(ResourceFilter.DataType.TEXT, ResourceFilter.Type.CONTAINS, "A_VLHV1_0", "fault.id")),
+                List.of(faultResultEntity1)),
+            Arguments.of(
+                resultMagnitudeEntity,
+                List.of(
+                    new ResourceFilter(ResourceFilter.DataType.TEXT, ResourceFilter.Type.CONTAINS, "SUBJECT_1", "limitViolations.subjectId")),
+                List.of(faultResultEntity1)),
+            Arguments.of(
+                resultMagnitudeEntity,
+                List.of(
+                    new ResourceFilter(ResourceFilter.DataType.TEXT, ResourceFilter.Type.CONTAINS, "SUBJECT_2", "limitViolations.subjectId")),
+                List.of(faultResultEntity1, faultResultEntity2, faultResultEntity3))
+        );
+    }
+
+    private Stream<Arguments> provideNotEqualFilters() {
+        return Stream.of(
+            Arguments.of(
+                resultMagnitudeEntity,
+                List.of(
+                    new ResourceFilter(ResourceFilter.DataType.NUMBER, ResourceFilter.Type.NOT_EQUAL, "45.3", "current")),
+                List.of(faultResultEntity2, faultResultEntity3)),
+            Arguments.of(
+                resultMagnitudeEntity,
+                List.of(
+                    new ResourceFilter(ResourceFilter.DataType.NUMBER, ResourceFilter.Type.NOT_EQUAL, "47", "current")),
+                List.of(faultResultEntity1, faultResultEntity2, faultResultEntity3)),
+            Arguments.of(
+                resultMagnitudeEntity,
+                List.of(
+                    new ResourceFilter(ResourceFilter.DataType.NUMBER, ResourceFilter.Type.NOT_EQUAL, "47.3", "current"),
+                    new ResourceFilter(ResourceFilter.DataType.NUMBER, ResourceFilter.Type.NOT_EQUAL, "49.3", "current")),
+                List.of(faultResultEntity1)),
+            Arguments.of(
+                resultMagnitudeEntity,
+                List.of(
+                    new ResourceFilter(ResourceFilter.DataType.NUMBER, ResourceFilter.Type.NOT_EQUAL, "2", "nbLimitViolations")),
+                List.of(faultResultEntity2))
+        );
+    }
+
+    private Stream<Arguments> provideNotEqualNestedFieldsFilters() {
+        return Stream.of(
+            //TODO FM need to fix it when we'll filter on limitViolations
```

Comment:
TODO ?

Reply:
This case is bugged but not used yet. I couldn't find why. I've let the tests because it can be useful later but I can remove them if you prefer.